docs/observabilityWalkthrough.md

```diff
@@ -0,0 +1,166 @@
+# Observability in Service Deployments
+
+Many Kubernetes deployments are composed of not just one, but many microservices, and this complexity is compounded by, for latency, scalability, and/or reliability concerns, that these microservices are often also deployed across multiple clusters as well.  This makes it hard to reason about the current state of any individual cluster -- and especially a collection of ones that all together constitute the workload.
+
+To help with this, Bedrock has a service introspection tool to provide better visibility the end to end deployment workflows. It integrates with the GitOps pipeline and service management that were setup in previous walkthrough guides. 
+
+The service introspection tool exists in two forms:
+1. Command line tool
+2. Web Dashboard
+
+The service introspection tool provides views into the current status of any change in the system, from continuous integration build to tracking the deployment of the container containing that commit in each of the downstream clusters consuming that container.
+
+The service introspection main components are:
+1. A Bedrock GitOps pipelines workflow. Currently supported in Azure DevOps
+2. An Azure Storage Table
+3. Service introspection command line tool and web dashboard
+
+The following diagram shows how the introspection tool integrates with the Azure DevOps pipelines in a Bedrock GitOps Workflow.
+![Service Introspection Tool](images/service-introspection-tool.png)
+
+The Web Dashboard is shown in the image below. In Bedrock, the status is displayed from newest to oldest, and we can see in the first line that a commit has triggered a container build that is currently deployed in the west cluster but that the east cluster hasn't yet been synchronized and is currently still running the previous version of the container.
+
+![Web Dashboard](images/service-introspection-dashboard.png)
+
+This walkthrough will cover how to set up introspection in your own deployments and how to use it to observe changes to your cluster.
+
+## Prerequisites
+This guideline assumes you have completed the following:
+
+1. Set up GitOps. Guideline: [A First Workload With Bedrock](./firstWorkload/README.md)
+2. Set up the HLD to Manifests pipeline. Guideline: [Setting up an HLD to Manifest pipeline](hldToManifestWalkthrough.md)
+3. Onboard a Service Repository. Guideline: [Service Management](services.md)
+
+
+## Setup an Azure Storage Table
+Service introspection tool needs a database to store the information about your
+pipelines, builds and deployments. Currently, service introspection supports
+storage in the form of an Azure Storage table. Follow the steps below to create
+it or use an existing one.
+
+### Create an Azure storage account
+
+**Option 1:**
+
+Use the `spk deployment onboard` command.
```

Comment:
We might explain a bit what this will do in terms of the storage accounts it creates and configuration of spk that it performs (if any).